py/picca/delta_extraction/expected_fluxes/dr16_expected_flux.py

```diff
@@ -625,184 +688,72 @@ def compute_var_stats(self, forests):
-
-                var_lss: float
-                Pixel variance due to the Large Scale Strucure
-
-                fudge: float
-                Fudge contribution to the pixel variance
-
-                Global arguments
-                ----------------
-                (defined only in the scope of function compute_var_stats):
-
-                var_delta: array of floats
-                Variance of the delta field
-
-                var2_delta: array of floats
-                Square of the variance of the delta field
-
-                index: int
-                Index with the selected wavelength bin
-
-                num_var_bins: int
-                Number of bins in which the pipeline variance values are split
-
-                var_pipe_values: array of floats
-                Value of the pipeline variance in pipeline variance bins
-
-                num_qso: array of ints
-                Number of quasars in each pipeline variance bin
-
-                Return
-                ------
-                chi2: float
-                The obtained chi2
-                """
-                variance = eta * var_pipe_values + var_lss + fudge * fudge_ref / var_pipe_values
-                chi2_contribution = (
-                    var_delta[index * num_var_bins:(index + 1) * num_var_bins] -
-                    variance)
-                weights = var2_delta[index * num_var_bins:(index + 1) *
-                                     num_var_bins]
-                w = num_qso[index * num_var_bins:(index + 1) *
-                            num_var_bins] > 100
-                return np.sum(chi2_contribution[w]**2 / weights[w])
-
-            minimizer = iminuit.Minuit(chi2,
+            leasts_squares.set_fit_bins(index)
+
+            minimizer = iminuit.Minuit(leasts_squares,
                                        name=("eta", "var_lss", "fudge"),
-                                       eta=1.,
-                                       var_lss=0.1,
-                                       fudge=1.)
+                                       eta=eta[index],
+                                       var_lss=var_lss[index],
+                                       fudge=fudge[index] / FUDGE_REF)
             minimizer.errors["eta"] = 0.05
+            minimizer.limits["eta"] = self.limit_eta
             minimizer.errors["var_lss"] = 0.05
+            minimizer.limits["var_lss"] = self.limit_var_lss
             minimizer.errors["fudge"] = 0.05
+            minimizer.limits["fudge"] = (0, None)
             minimizer.errordef = 1.
             minimizer.print_level = 0
-            minimizer.limits["eta"] = self.limit_eta
-            minimizer.limits["var_lss"] = self.limit_var_lss
-            minimizer.limits["fudge"] = (0, None)
+            minimizer.fixed["eta"] = "eta" not in self.fit_variance_functions
+            minimizer.fixed[
+                "var_lss"] = "var_lss" not in self.fit_variance_functions
```

Comment:
is this not only evaluating to true for var_lss by default? See comment further up.
Shouldn't this variable be changed in the Fixed... classes? See comments there.